terracotta/drivers/terracotta_driver.py

```diff
@@ -169,7 +169,13 @@ def get_metadata(self, keys: ExtendedKeysType) -> Dict[str, Any]:
 
             path = squeeze(dataset.values())
             metadata = self.compute_metadata(path, max_shape=self.LAZY_LOADING_MAX_SHAPE)
-            self.insert(keys, path, metadata=metadata)
+
+            try:
+                self.insert(keys, path, metadata=metadata)
+            except exceptions.DatabaseNotWritableError as exc:
+                raise exceptions.DatabaseNotWritableError(
+                    "Lazy loading requires a writable database"
+                ) from exc
```

Comment:
Maybe, if you have a convincing use case. I don't really want to encourage lazy loading too much, especially not for production contexts. So the functionality is limited by design.

Reply:
> And also, @kiksekage, the STAC driver you were working on, required lazy loading without writable, right?

Yeah, the choice was either a) maintain a local DB for metadata calculation results or b) only do on-the-fly metadata calculations, and b) seemed nicer by simply not having to maintain a local DB. This might still change though! 
I like the idea of a warning, and currently I guess it would only be raised when lazy loading on a sqlite-remote metastore. Since that didn't work actually work in the codebase before, it might be a small enough niche that it shouldn't affect too many users?

Reply:
I suggest we keep it as it is for now and discuss this in a separate issue.